go-controller/pkg/ovn/master.go

```diff
@@ -776,35 +784,23 @@ func (oc *Controller) ensureNodeLogicalNetwork(nodeName string, hostSubnets []*n
 		return err
 	}
 
-	// Add our cluster TCP and UDP load balancers to the node switch
-	if oc.TCPLoadBalancerUUID == "" {
-		return fmt.Errorf("TCP cluster load balancer not created")
-	}
-	stdout, stderr, err = util.RunOVNNbctl("set", "logical_switch", nodeName, "load_balancer="+oc.TCPLoadBalancerUUID)
-	if err != nil {
-		klog.Errorf("Failed to set logical switch %v's load balancer, stdout: %q, stderr: %q, error: %v", nodeName, stdout, stderr, err)
-		return err
-	}
-
-	if oc.UDPLoadBalancerUUID == "" {
-		return fmt.Errorf("UDP cluster load balancer not created")
-	}
-	stdout, stderr, err = util.RunOVNNbctl("add", "logical_switch", nodeName, "load_balancer", oc.UDPLoadBalancerUUID)
-	if err != nil {
-		klog.Errorf("Failed to add logical switch %v's load balancer, stdout: %q, stderr: %q, error: %v", nodeName, stdout, stderr, err)
-		return err
-	}
-
+	// Add our cluster load balancers per protocol to the node switch
+	protocols := []v1.Protocol{v1.ProtocolTCP, v1.ProtocolUDP}
 	if oc.SCTPSupport {
-		if oc.SCTPLoadBalancerUUID == "" {
-			return fmt.Errorf("SCTP cluster load balancer not created")
+		protocols = append(protocols, v1.ProtocolSCTP)
+	}
+	for _, p := range protocols {
+		lbUUID, err := loadbalancer.GetOVNKubeLoadBalancer(p)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to get OVN load balancer for protocol %s", p)
 		}
-		stdout, stderr, err = util.RunOVNNbctl("add", "logical_switch", nodeName, "load_balancer", oc.SCTPLoadBalancerUUID)
+		stdout, stderr, err = util.RunOVNNbctl("set", "logical_switch", nodeName, "load_balancer="+lbUUID)
```

Comment:
previous code uses `set` for the TCP lb and `add` for the UDP one, are both the same?